### PR TITLE
feat: Add interface comments

### PR DESCRIPTION
**Pull request by kkumar-gcc**
Closes: https://github.com/goravel/goravel/issues/128

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

File: contracts/cache/cache.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ type Cache interface {
 type Driver interface {
 	//Add Driver an item in the cache if the key does not exist.
 	Add(key string, value any, t time.Duration) bool
+	//Decrement Decrements the value of an item in the cache.
```
Comment (hwbrzzl) [inline contracts/cache/cache.go:18]:
```suggestion
	//Decrement decrements the value of an item in the cache.
```

File: contracts/cache/cache.go [hunk 2]
```diff
@@ -24,13 +25,19 @@ type Driver interface {
...
 	GetBool(key string, def ...bool) bool
+	// GetInt retrieves an item from the cache by key as an integer.
 	GetInt(key string, def ...int) int
+	// GetInt64 retrieves an item from the cache by key as a 64-bit integer.
 	GetInt64(key string, def ...int64) int64
+	// GetString retrieves an item from the cache by key as a string.
 	GetString(key string, def ...string) string
 	//Has Check an item exists in the cache.
 	Has(key string) bool
+	//Increment Increments the value of an item in the cache.
```
Comment (hwbrzzl) [inline contracts/cache/cache.go:38]:
```suggestion
	//Increment increments the value of an item in the cache.
```

File: contracts/cache/cache.go [hunk 3]
```diff
@@ -24,13 +25,19 @@ type Driver interface {
...
 	Increment(key string, value ...int) (int, error)
+	//Lock Get a lock instance.
```
Comment (hwbrzzl) [inline contracts/cache/cache.go:40]:
```suggestion
	//Lock get a lock instance.
```

File: contracts/cache/cache.go [hunk 4]
```diff
@@ -40,13 +47,18 @@ type Driver interface {
 	Remember(key string, ttl time.Duration, callback func() (any, error)) (any, error)
 	//RememberForever Get an item from the cache, or execute the given Closure and store the result forever.
 	RememberForever(key string, callback func() (any, error)) (any, error)
+	//WithContext returns a new Cache instance with the given context.
 	WithContext(ctx context.Context) Driver
 }
 
 //go:generate mockery --name=Lock
 type Lock interface {
+	//Block Attempt to acquire the lock for the given number of seconds.
```
Comment (hwbrzzl) [inline contracts/cache/cache.go:56]:
```suggestion
	//Block attempt to acquire the lock for the given number of seconds.
```

File: contracts/cache/cache.go [hunk 5]
```diff
@@ -40,13 +47,18 @@ type Driver interface {
...
 	Block(t time.Duration, callback ...func()) bool
+	//Get Attempt to acquire the lock.
```
Comment (hwbrzzl) [inline contracts/cache/cache.go:58]:
```suggestion
	//Get attempt to acquire the lock.
```

File: contracts/cache/cache.go [hunk 6]
```diff
@@ -40,13 +47,18 @@ type Driver interface {
...
 	Get(callback ...func()) bool
+	//Release the lock.
 	Release() bool
+	//ForceRelease Releases the lock in disregard of ownership.
```
Comment (hwbrzzl) [inline contracts/cache/cache.go:62]:
```suggestion
	//ForceRelease releases the lock in disregard of ownership.
```

File: contracts/console/artisan.go [hunk 7]
```diff
@@ -2,15 +2,15 @@ package console
 
 //go:generate mockery --name=Artisan
 type Artisan interface {
-	//Register commands.
+	// Register commands.
 	Register(commands []Command)
 
-	//Call Run an Artisan console command by name.
+	// Call Run an Artisan console command by name.
```
Comment (hwbrzzl) [inline contracts/console/artisan.go:8]:
```suggestion
	// Call run an Artisan console command by name.
```

File: contracts/console/artisan.go [hunk 8]
```diff
@@ -2,15 +2,15 @@ package console
...
 	Call(command string)
 
-	//CallAndExit Run an Artisan console command by name and exit.
+	// CallAndExit Run an Artisan console command by name and exit.
```
Comment (hwbrzzl) [inline contracts/console/artisan.go:11]:
```suggestion
	// CallAndExit run an Artisan console command by name and exit.
```

File: contracts/console/command.go [hunk 9]
```diff
@@ -5,27 +5,38 @@ import (
 )
 
 type Command interface {
-	//Signature The name and signature of the console command.
+	// Signature set the unique signature for the command.
 	Signature() string
-	//Description The console command description.
+	// Description The console command description.
```
Comment (hwbrzzl) [inline contracts/console/command.go:10]:
```suggestion
	// Description the console command description.
```

File: contracts/console/command.go [hunk 10]
```diff
@@ -5,27 +5,38 @@ import (
...
 	Description() string
-	//Extend The console command extend.
+	// Extend The console command extend.
```
Comment (hwbrzzl) [inline contracts/console/command.go:12]:
```suggestion
	// Extend the console command extend.
```

File: contracts/console/command.go [hunk 11]
```diff
@@ -5,27 +5,38 @@ import (
...
 	Extend() command.Extend
-	//Handle Execute the console command.
+	// Handle Execute the console command.
```
Comment (hwbrzzl) [inline contracts/console/command.go:14]:
```suggestion
	// Handle execute the console command.
```

File: contracts/console/command/command.go [hunk 12]
```diff
@@ -18,6 +18,7 @@ type Extend struct {
 }
 
 type Flag interface {
+	// Type Gets a flag type.
```
Comment (hwbrzzl) [inline contracts/console/command/command.go:21]:
```suggestion
	// Type gets a flag type.
```

File: contracts/database/factory/factory.go [hunk 13]
```diff
@@ -1,9 +1,11 @@
 package factory
 
 type Factory interface {
+	// Definition Defines the model's default state.
```
Comment (hwbrzzl) [inline contracts/database/factory/factory.go:4]:
```suggestion
	// Definition defines the model's default state.
```

File: contracts/database/factory/factory.go [hunk 14]
```diff
@@ -1,9 +1,11 @@
...
 	Definition() map[string]any
 }
 
 type Model interface {
+	// Factory Creates a new factory instance for the model.
```
Comment (hwbrzzl) [inline contracts/database/factory/factory.go:9]:
```suggestion
	// Factory creates a new factory instance for the model.
```

File: contracts/database/orm/factory.go [hunk 15]
```diff
@@ -2,8 +2,12 @@ package orm
 
 //go:generate mockery --name=Factory
 type Factory interface {
+	// Count Sets the number of models that should be generated.
```
Comment (hwbrzzl) [inline contracts/database/orm/factory.go:5]:
```suggestion
	// Count sets the number of models that should be generated.
```

File: contracts/database/orm/factory.go [hunk 16]
```diff
@@ -2,8 +2,12 @@ package orm
...
 	Count(count int) Factory
+	// Create Creates a model and persists it to the database.
```
Comment (hwbrzzl) [inline contracts/database/orm/factory.go:7]:
```suggestion
	// Create creates a model and persists it to the database.
```

File: contracts/database/orm/factory.go [hunk 17]
```diff
@@ -2,8 +2,12 @@ package orm
...
 	Create(value any, attributes ...map[string]any) error
+	// CreateQuietly Creates a model and persists it to the database without firing any model events.
```
Comment (hwbrzzl) [inline contracts/database/orm/factory.go:9]:
```suggestion
	// CreateQuietly creates a model and persists it to the database without firing any model events.
```

File: contracts/database/orm/factory.go [hunk 18]
```diff
@@ -2,8 +2,12 @@ package orm
...
 	CreateQuietly(value any, attributes ...map[string]any) error
+	// Make Creates a model and returns it, but does not persist it to the database.
```
Comment (hwbrzzl) [inline contracts/database/orm/factory.go:11]:
```suggestion
	// Make creates a model and returns it, but does not persist it to the database.
```

File: contracts/database/orm/observer.go [hunk 19]
```diff
@@ -1,15 +1,26 @@
 package orm
 
 type Observer interface {
+	// Retrieved Called when the model is retrieved from the database.
```
Comment (hwbrzzl) [inline contracts/database/orm/observer.go:4]: Ditto
```suggestion
	// Retrieved called when the model is retrieved from the database.
```

File: contracts/database/orm/orm.go [hunk 20]
```diff
@@ -7,89 +7,157 @@ import (
 
 //go:generate mockery --name=Orm
 type Orm interface {
+	// Connection gets an Orm instance from the connection pool.
 	Connection(name string) Orm
+	// DB gets the underlying database connection.
 	DB() (*sql.DB, error)
+	// Query gets a new query builder instance.
 	Query() Query
+	// Factory Gets a new factory instance for the given model name.
```
Comment (hwbrzzl) [inline contracts/database/orm/orm.go:16]:
```suggestion
	// Factory gets a new factory instance for the given model name.
```

File: contracts/database/orm/orm.go [hunk 21]
```diff
@@ -7,89 +7,157 @@ import (
...
 	Factory() Factory
+	// Observe Observer Registers an observer with the Orm.
```
Comment (hwbrzzl) [inline contracts/database/orm/orm.go:18]:
```suggestion
	// Observe registers an observer with the Orm.
```

File: contracts/database/orm/orm.go [hunk 22]
```diff
@@ -7,89 +7,157 @@ import (
...
 	Rollback() error
 }
 
 //go:generate mockery --name=Query
 type Query interface {
+	// Association gets an association instance by name.
 	Association(association string) Association
+	// Begin begins a new transaction
 	Begin() (Transaction, error)
+	// The Driver gets the driver for the query.
```
Comment (hwbrzzl) [inline contracts/database/orm/orm.go:41]:
```suggestion
	// Driver gets the driver for the query.
```

File: contracts/database/orm/orm.go [hunk 23]
```diff
@@ -7,89 +7,157 @@ import (
...
 	FirstOrCreate(dest any, conds ...any) error
+	// FirstOr finds the first record that matches the given conditions or
+	// execute the callback and return its result if no record is found.
 	FirstOr(dest any, callback func() error) error
+	// FirstOrFail finds the first record that matches the given conditions or throws an error.
 	FirstOrFail(dest any) error
+	// FirstOrNew finds the first record that matches the given conditions or
+	// return a new instance of the model initialized with those attributes.
 	FirstOrNew(dest any, attributes any, values ...any) error
+	// ForceDelete deletes records matching given conditions.
```
Comment (hwbrzzl) [inline contracts/database/orm/orm.go:72]:
```suggestion
	// ForceDelete forces delete records matching given conditions.
```

File: contracts/foundation/application.go [hunk 24]
```diff
@@ -7,13 +7,22 @@ import (
 //go:generate mockery --name=Application
 type Application interface {
 	Container
+	// Boot Register and bootstrap configured service providers.
```
Comment (hwbrzzl) [inline contracts/foundation/application.go:10]:
```suggestion
	// Boot register and bootstrap configured service providers.
```

File: contracts/foundation/container.go [hunk 25]
```diff
@@ -24,31 +24,58 @@ import (
 )
 
 type Container interface {
+	// Bind Registers a binding with the container.
```
Comment (hwbrzzl) [inline contracts/foundation/container.go:27]: Ditto
```suggestion
	// Bind tegisters a binding with the container.
```

File: contracts/grpc/grpc.go [hunk 26]
```diff
@@ -8,9 +8,14 @@ import (
 
 //go:generate mockery --name=Grpc
 type Grpc interface {
+	// Run starts the gRPC server.
 	Run(host ...string) error
+	// Server gets the gRPC server instance.
 	Server() *grpc.Server
+	// Client gets the gRPC client instance.
 	Client(ctx context.Context, name string) (*grpc.ClientConn, error)
+	// UnaryServerInterceptors gets the gRPC server interceptors.
```
Comment (hwbrzzl) [inline contracts/grpc/grpc.go:17]:
```suggestion
	// UnaryServerInterceptors sets the gRPC server interceptors.
```

File: contracts/grpc/grpc.go [hunk 27]
```diff
@@ -8,9 +8,14 @@ import (
...
 	UnaryServerInterceptors([]grpc.UnaryServerInterceptor)
+	// UnaryClientInterceptorGroups gets the gRPC client interceptor groups.
```
Comment (hwbrzzl) [inline contracts/grpc/grpc.go:19]:
```suggestion
	// UnaryClientInterceptorGroups sets the gRPC client interceptor groups.
```

File: contracts/http/rate_limiter.go [hunk 28]
```diff
@@ -2,12 +2,17 @@ package http
 
 //go:generate mockery --name=RateLimiter
 type RateLimiter interface {
+	// For Register a new rate limiter.
```
Comment (hwbrzzl) [inline contracts/http/rate_limiter.go:5]: Ditto
```suggestion
	// For register a new rate limiter.
```

File: contracts/http/request.go [hunk 29]
```diff
@@ -9,56 +9,84 @@ import (
...
 
+	// Validate performs request data validation using specified rules and options.
 	Validate(rules map[string]string, options ...validation.Option) (validation.Validator, error)
+	// ValidateRequest validates the request data against a pre-defined FormRequest structure
+	// and returns validation errors, if any.
 	ValidateRequest(request FormRequest) (validation.Errors, error)
 }
 
 type FormRequest interface {
+	// Authorize Determine if the user is authorized to make this request.
```
Comment (hwbrzzl) [inline contracts/http/request.go:82]: Ditto
```suggestion
	// Authorize determine if the user is authorized to make this request.
```

File: contracts/http/response.go [hunk 30]
```diff
@@ -14,44 +14,70 @@ type Response interface {
 
 //go:generate mockery --name=ContextResponse
 type ContextResponse interface {
+	// Data Write the given data to the response.
```
Comment (hwbrzzl) [inline contracts/http/response.go:17]:
```suggestion
	// Data write the given data to the response.
```

File: contracts/http/response.go [hunk 31]
```diff
@@ -14,44 +14,70 @@ type Response interface {
...
 	View() ResponseView
+	// Writer returns the underlying http.ResponseWriter associated with the response.
 	Writer() http.ResponseWriter
+	// Flush flushes any buffered data to the client.
 	Flush()
 }
 
 //go:generate mockery --name=ResponseStatus
 type ResponseStatus interface {
+	// Data Write the given data to the Response.
```
Comment (hwbrzzl) [inline contracts/http/response.go:48]:
```suggestion
	// Data write the given data to the Response.
```

File: contracts/http/response.go [hunk 32]
```diff
@@ -14,44 +14,70 @@ type Response interface {
...
 	Data(contentType string, data []byte) Response
+	// Json sends a JSON Response with the specified data object.
 	Json(obj any) Response
+	// String writes a string Response with the specified format and values.
 	String(format string, values ...any) Response
 }
 
 //go:generate mockery --name=ResponseSuccess
 type ResponseSuccess interface {
+	// Data Write the given data to the Response.
```
Comment (hwbrzzl) [inline contracts/http/response.go:58]:
```suggestion
	// Data write the given data to the Response.
```

File: contracts/log/log.go [hunk 33]
```diff
@@ -68,22 +81,26 @@ type Writer interface {
...
 
 //go:generate mockery --name=Entry
 type Entry interface {
+	// Context returns the context of the entry.
 	Context() context.Context
+	// Level returns the level of the entry.
 	Level() Level
+	// Time returns the timestamp of the entry.
 	Time() time.Time
+	// The Message returns the message of the entry.
```
Comment (hwbrzzl) [inline contracts/log/log.go:104]:
```suggestion
	// Message returns the message of the entry.
```

File: contracts/schedule/event.go [hunk 34]
```diff
@@ -2,33 +2,62 @@ package schedule
 
 //go:generate mockery --name=Event
 type Event interface {
+	// At Schedule the event to run at the specified time.
```
Comment (hwbrzzl) [inline contracts/schedule/event.go:5]: Ditto
```suggestion
	// At schedule the event to run at the specified time.
```

File: contracts/testing/testing.go [hunk 35]
```diff
@@ -8,25 +8,36 @@ import (
 )
 
 type Testing interface {
+	// Docker Get the Docker instance.
```
Comment (hwbrzzl) [inline contracts/testing/testing.go:11]: Ditto
```suggestion
	// Docker get the Docker instance.
```

File: contracts/validation/validation.go [hunk 36]
```diff
@@ -4,33 +4,48 @@ type Option func(map[string]any)
 
 //go:generate mockery --name=Validation
 type Validation interface {
+	// Make Create a new validator instance.
```
Comment (hwbrzzl) [inline contracts/validation/validation.go:7]: Ditto
```suggestion
	// Make create a new validator instance.
```

**Review by hwbrzzl — COMMENTED** [review]
Great work, Bro!

File: contracts/cache/cache.go [hunk 37]
```diff
@@ -15,38 +15,50 @@ type Cache interface {
...
-	//Pull Retrieve an item from the cache and delete it.
+	//Pull retrieve an item from the cache and delete it.
 	Pull(key string, def ...any) any
-	//Remember Get an item from the cache, or execute the given Closure and store the result.
+	// Remember gets an item from the cache, or execute the given Closure and store the result.
 	Remember(key string, ttl time.Duration, callback func() (any, error)) (any, error)
-	//RememberForever Get an item from the cache, or execute the given Closure and store the result forever.
+	//RememberForever get an item from the cache, or execute the given Closure and store the result forever.
 	RememberForever(key string, callback func() (any, error)) (any, error)
+	//WithContext returns a new Cache instance with the given context.
```
Comment (devhaozi) [inline contracts/cache/cache.go:50]: Add a space for each after `//`

File: contracts/filesystem/storage.go [hunk 38]
```diff
@@ -8,47 +8,82 @@ import (
...
+	// Exists determines if a file exists.
 	Exists(file string) bool
+	// Files gets all the files from the given directory.
 	Files(path string) ([]string, error)
+	// Get gets the contents of a file.
 	Get(file string) (string, error)
+  // GetBytes gets the contents of a file as a byte array.
 	GetBytes(file string) ([]byte, error)
+  // LastModified gets the file's last modified time.
 	LastModified(file string) (time.Time, error)
```
Comment (devhaozi) [inline contracts/filesystem/storage.go:38]: pls format this.

**Review by hwbrzzl — APPROVED** [review]
LGTM

**Pull request merged**